express-notes-api/index.js

```diff
@@ -0,0 +1,113 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const useJson = express.json();
+const jsonData = require('./data.json');
+let resArr = [];
+
+app.use(useJson);
+
+const pushJsonIntoArray = () => {
+  const eachNoteArr = Object.entries(jsonData.notes);
+  for (let i = 0; i < eachNoteArr.length; i++) {
+    resArr.push(eachNoteArr[i][1]);
+  }
+};
+
+app.get('/api/notes', (req, res) => {
+  resArr = [];
+  pushJsonIntoArray();
+  res.status(200).send(resArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  resArr = [];
+  pushJsonIntoArray();
+  const userRequestId = parseInt(req.params.id);
+  if (userRequestId <= 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (userRequestId < jsonData.nextId) {
+    for (let i = 0; i < resArr.length; i++) {
+      if (resArr[i].id === userRequestId) {
+        res.status(200).send(resArr[i]);
+      }
+    }
+  } else if (!userRequestId) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+
+  } else {
+    res.status(404).send({ error: `cannot find note with id ${userRequestId}` });
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  const content = req.body;
+
+  if (Object.keys(content).length === 0) {
+    res.status(400).send({ error: 'Content is a required field' });
+
+  } else if (content) {
+    jsonData.notes[jsonData.nextId] = content;
+    jsonData.notes[jsonData.nextId].id = jsonData.nextId;
+    jsonData.nextId++;
+    fs.writeFile('data.json', JSON.stringify(jsonData, null, 2), err => {
+      if (err) {
+        console.error(err);
+        res.status(500).send({ error: 'An unexpected error occured' });
+
+      }
+    });
+    const currentId = jsonData.nextId - 1;
+    res.status(201).send(jsonData.notes[currentId]);
+  }
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  const userRequestId = parseInt(req.params.id);
+  if (userRequestId <= 0) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else if (userRequestId < jsonData.nextId) {
+    delete jsonData.notes[userRequestId];
+    res.sendStatus(204);
+    fs.writeFile('data.json', JSON.stringify(jsonData, null, 2), err => {
+      if (err) {
+        console.error(err);
+        res.status(500).send({ error: 'An unexpected error occured' });
+
+      }
+    });
+  } else if (!userRequestId) {
+    res.status(400).send({ error: 'id must be a positive integer' });
+  } else {
+    res.status(404).send({ error: `cannot find note with id ${userRequestId}` });
+  }
+});
+
+app.put('/api/notes/:id', (req, res) => {
+
+  const userRequestId = parseInt(req.params.id);
+  const requestBody = req.body;
+
+  if (userRequestId <= 0 || Object.keys(requestBody).length === 0) {
```

Comment:
Again, the `requestBody` may contain keys other than `content`

Reply:
done